libraries/lib-audio-io/AudioIO.cpp

```diff
@@ -2036,8 +2037,15 @@ bool AudioIO::ProcessPlaybackSlices(
 
                auto& buffer = mProcessingBuffers[iBuffer++];
                const auto offset = buffer.size();
+               const auto toConsume = std::min(
```

Comment:
What are these new lines fixing ? It looks like we are now ensuring that no new memory is allocated in that function. This comes at the risk, though, that we provide less than `frames` samples, if for some reason the capacity of the buffer is insufficient. Wouldn't that be worse ?

I wonder if this is because of the initial playback click that we hear in @dozzzzer 's video above. I suspect this not to be due to buffer allocation, but simply because he is playing a sine wave without any sort of smoothing envelope, and it's then completely normal to hear this click.

Reply:
>  It looks like we are now ensuring that no new memory is allocated in that function. 

Yes

> This comes at the risk, though, that we provide less than frames samples

Memory allocations are not better. The disadvantage of this code is that it's not capable to handle very large latencies (usually > 250ms).

> I wonder if this change is motivated by the initial playback click that we hear in @dozzzzer 's video above

No, I did that in response to Paul's comment above